src/SFML/Graphics/Sprite.cpp

```diff
@@ -131,36 +122,35 @@ void Sprite::draw(RenderTarget& target, RenderStates states) const
     states.transform *= getTransform();
     states.texture        = m_texture;
     states.coordinateType = CoordinateType::Pixels;
-    target.draw(m_vertices.data(), m_vertices.size(), PrimitiveType::TriangleStrip, states);
-}
-
-
-////////////////////////////////////////////////////////////
-void Sprite::updatePositions()
-{
-    const FloatRect bounds = getLocalBounds();
 
-    m_vertices[0].position = Vector2f(0, 0);
-    m_vertices[1].position = Vector2f(0, bounds.height);
-    m_vertices[2].position = Vector2f(bounds.width, 0);
-    m_vertices[3].position = Vector2f(bounds.width, bounds.height);
+    target.draw(m_vertices.data(), m_vertices.size(), PrimitiveType::TriangleStrip, states);
 }
 
 
 ////////////////////////////////////////////////////////////
-void Sprite::updateTexCoords()
+void Sprite::updateVertices()
 {
-    const FloatRect convertedTextureRect(m_textureRect);
-
-    const float left   = convertedTextureRect.left;
-    const float right  = left + convertedTextureRect.width;
-    const float top    = convertedTextureRect.top;
-    const float bottom = top + convertedTextureRect.height;
-
-    m_vertices[0].texCoords = Vector2f(left, top);
-    m_vertices[1].texCoords = Vector2f(left, bottom);
-    m_vertices[2].texCoords = Vector2f(right, top);
-    m_vertices[3].texCoords = Vector2f(right, bottom);
+    assert(m_textureRect.width >= 0);
+    assert(m_textureRect.height >= 0);
+
+    const auto width  = static_cast<float>(m_textureRect.width);
+    const auto height = static_cast<float>(m_textureRect.height);
+    const auto left   = static_cast<float>(m_textureRect.left);
+    const auto top    = static_cast<float>(m_textureRect.top);
+    const auto right  = float{left + width};
+    const auto bottom = float{top + height};
```

Comment:
I have to admit that @ChrisThrasher's point regarding using `T{...}` instead of `static_cast<T>(...)` was great, I was wrong in doubting it.

I can now easily tell that the first four conversions are lossy, while the last two are lossless.